framework/src/main/java/org/checkerframework/framework/type/DefaultAnnotatedTypeFormatter.java

```diff
@@ -385,26 +385,37 @@ public String visitNull(AnnotatedNullType type, Set<AnnotatedTypeMirror> visitin
         @Override
         public String visitWildcard(AnnotatedWildcardType type, Set<AnnotatedTypeMirror> visiting) {
             StringBuilder sb = new StringBuilder();
-            sb.append(
-                    annoFormatter.formatAnnotationString(
-                            type.getAnnotationsField(), currentPrintInvisibleSetting));
-            sb.append("?");
-            if (!visiting.contains(type)) {
+            if (type.isUninferredTypeArgument()) {
+                sb.append(
+                        annoFormatter.formatAnnotationString(
+                                type.getAnnotationsField(), currentPrintInvisibleSetting));
+                sb.append(" /*INFERENCE FAILED ");
```

Comment:
What is the motivation for the comment syntax?
How about using:

```java
if (type.isUninferredTypeArgument()) {
    sb.append("INFERENCE FAILED for: ");
}
```

as the first thing and leaving all the rest unchanged?

Reply:
We add `/*DECL*/` for declarations, so I was trying to be consistent with that.  It's kind of nice that it is a comment so that the printed type is still legal Java.  It's also good to make it clear that the `INFERENCE FAILED` is a comment and not part of the type.  I will change it to ` /*INFERENCE FAILED for: `.

Reply:
Having the comment before the type instead of between the annotations and the type would make it more readable, at least to me. So I would rather add

```java
if (type.isUninferredTypeArgument()) {
    sb.append("/* INFERENCE FAILED for: */");
}
```

and then keep the output of the type unchanged.

Reply:
So the output would be:
```
  /*INFERENCE FAILED for:*/ @Initialized @NonNull ? extends @Initialized @NonNull List<@Initialized @NonNull String>
```